### ISSUE TITLE
Error finding retriever PATH in Windows

### ISSUE THREAD

**Issue by Pakillo** [ropensci/rdataretriever#117]
Hi,

First of all, thanks for the package! Really useful.

I am having problems getting the package to work. I have installed the latest retriever software from here (http://www.data-retriever.org/). It adds itself to the path and works perfectly from the terminal. However, the R package cannot find the software (using latest Rstudio).

When I run 
`library(rdataretriever)`

I get:

The retriever is not on your path and may not be installed. Please upgrade to the most recent version of the Data Retriever, which will automatically add itself to the path http://www.data-retriever.org/#install

I have been investigating a little bit and I suspect the problem might be in the `check_for_retriever` function (https://github.com/ropensci/rdataretriever/blob/master/R/rdataretriever.R#L293) which is run on package load.

I have seen it tries to add new paths for Anaconda, miniconda, etc. However, that doesn't work in my system.
I have tried just adding the path for retriever in my computer: just running `Sys.setenv(PATH = paste(Sys.getenv('PATH'), ';', "C:\\DataRetriever", sep = ''))` and it works! 
So, I wonder if that anaconda stuff is coming from previous software configurations?

(Note I have installed the retriever software in C:/ (root). The default location for the installer would in the 'Program files' folder in most Windows systems; so that would have to be adapted).

Hope this info helps to get the problem fixed! Please let me know if I'm doing something wrong :)

My session info:

R version 3.4.1 (2017-06-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rdataretriever_1.0.0

loaded via a namespace (and not attached):
[1] compiler_3.4.1 tools_3.4.1

**Comment by henrykironde** [comment]
Hi @Pakillo thank you for reporting this issue. It is nice to hear that the tool is able to help you with managing your data. If you are running `Windows` and `Rstudio`, `Rstudio` tends to fail at getting the paths correct. I remember I had to change some settings in `Rstudio`. When it comes to Anaconda, there may be complications. I am going to test the issue further and come up with a more concrete solution.

**Comment by Pakillo** [comment]
Thank you @henrykironde 
In case it helps, this is the hack I made to make it work: https://github.com/Pakillo/rdataretriever/commit/ec82a811df436f79d54e3861fc3dd96ed3f760c2#diff-f9db74bd7ef4887b6a85c438844b8d5dR301

**Comment by ethanwhite** [comment]
Hi @Pakillo - Thanks so much for reporting this! As @henrykironde said the Windows-RStudio combo is definitely a little fragile, which is why the code you've been modifying is there. Just so I fully understand things - am I correct that you installed the main Retriever using the Windows installer? If so can you also tell us which version of Windows you're running. The installer based installation path additions used to work with RStudio, so I'm just trying to understand what's changed.

The retriever can be installed in several ways that all result it in being installed in different places. Some of those ways put it in the conda/anaconda folders that you noted. The fix here is to basically add more folders to that list to cover the standard installer based locations since RStudio is apparently breaking those paths now. Since they aren't in the home directory we'll need to tweak the current setup a bit so that the appending of the home directory happens only for the anaconda paths.

@Pakillo - If you'd be interested in trying to PR a fix for this we'd be happy to help out as you go. We're also happy to just fix it, but since you've already done the hard work of figuring out the issue we'd love to have you contribute if you're interested.

**Comment by Pakillo** [comment]
Hi @ethanwhite !

> am I correct that you installed the main Retriever using the Windows installer?

Yes, I used the exe file for Windows from http://www.data-retriever.org/#install (actual installer: https://github.com/weecology/retriever/releases/download/v2.0.0/RetrieverSetup.exe)

> If so can you also tell us which version of Windows you're running

Sure, Windows 7 Professional (64 bits)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

> The installer based installation path additions used to work with RStudio

The retriever software adds itself to the path and works perfectly from the terminal. The problem is with Rstudio, which didn't get the complete path, as explained here: http://stackoverflow.com/questions/31121645/rstudio-shows-a-different-path-variable. So, if I ran `Sys.getenv("PATH")`, DataRetriever did not appear there.

The funny thing is that now, after using the hack above, Rstudio is giving me the whole PATH:
```
> Sys.getenv("PATH")
[1] "C:\\R\\R-3.4.1\\bin\\x64;c:\\Rtools\\bin;c:\\Rtools\\mingw_32\\bin;C:\\ProgramData\\Oracle\\Java\\javapath;C:\\Program Files\\ImageMagick-7.0.5-Q16;C:\\Program Files (x86)\\AutoFirma\\AutoFirma;C:\\Program Files (x86)\\MiKTeX 2.9\\miktex\\bin\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\PuTTY\\;C:\\Program Files (x86)\\Skype\\Phone\\;C:\\DataRetriever;C:\\Program Files\\Intel\\WiFi\\bin\\;C:\\Program Files\\Common Files\\Intel\\WirelessCommon\\;C:\\Program Files (x86)\\MiKTeX 2.9\\miktex\\bin\\;C:\\Users\\FRS\\AppData\\Roaming\\MiKTeX\\2.9\\miktex\\bin\\;;.;;.;"
```

Looks like Rstudio 'remembers' this stuff after the first time, as suggested in the [discussion](https://stackoverflow.com/questions/31121645/rstudio-shows-a-different-path-variable) at StackOverflow... Too confusing.

What I know for certain is that rdataretriever was not able to find the software after installation. And the problem was in Rstudio not getting the whole path at first.

In summary, I think the hack of adding 'possible pathes' as you currently have is OK, but it would need to include at least `Program files`, `Program files (x86)`, and probably also the root (C:), as these are the most likely folders to be installed. In my system, the default folder for installation was Program files (x86). 

However, that hack will not work if people install the software elsewhere. So ideally the best solution would then be to somehow being able to read the Windows path and detect where retriever has been installed. Or, alternatively, encourage people to not change the destination folder when installing the software.

Hope all this helps... I'm happy to submit a PR if you find that useful. Or leave you fix it, as you prefer!

Cheers

**Comment by ethanwhite** [comment]
Sorry for the delay in getting back to you @Pakillo. Too much travel.

> In summary, I think the hack of adding 'possible pathes' as you currently have is OK, but it would need to include at least Program files, Program files (x86), and probably also the root (C:), as these are the most likely folders to be installed. In my system, the default folder for installation was Program files (x86).

I agree that adding these three directories to the list would be a good temporary solution.

> However, that hack will not work if people install the software elsewhere. So ideally the best solution would then be to somehow being able to read the Windows path and detect where retriever has been installed. Or, alternatively, encourage people to not change the destination folder when installing the software.

We spent a fair bit of time trying to figure out how to do that in a way that was robust to RStudio, but never figured anything out. If you know of something that would certainly be ideal. Agree that a doc addition about not changing the default directory would be good (though I suspect that people using the installer and changing the default installation directory will be relatively rare).

> Hope all this helps... I'm happy to submit a PR if you find that useful. Or leave you fix it, as you prefer!

A PR would be great if you have the time and energy to contribute.

**Comment by Pakillo** [comment]
Ok, pull request sent! Feel free to change anything as you see fit.

I agree that most people will install to the default folder (i.e. `Program Files` or `Program Files (x86)`). 

As I said this hack fixed the problem in my system, but I'm not sure the hack will work everywhere (e.g. in Spanish Windows, the installation folder is 'C:/Archivos de Programa/'). I'll let you know if I come across more robust ways to find the actual path to the retriever software in Windows. 

Cheers

**Comment by ethanwhite** [comment]
Fixed by #118. Thanks again @Pakillo!

**Comment by Pakillo** [comment]
You're welcome! Thanks to you for fixing up everything, and releasing the software in the first place!
Cheers
